evm/hyperliquid.js

```diff
@@ -92,28 +75,22 @@ async function sendRequest(action, signature, nonce, isMainnet) {
     return result;
 }
 
-async function updateBlockSize(privateKey, useBig, network = 'mainnet') {
+async function updateBlockSize(privateKey, useBig, network = 'mainnet', chain) {
     validateParameters({ isValidPrivateKey: { privateKey } });
 
     const wallet = new Wallet(privateKey);
     const isMainnet = network.toLowerCase() === 'mainnet';
 
     const action = { type: 'evmUserModify', usingBigBlocks: useBig };
     const nonce = Date.now();
-    const signature = await signL1Action(wallet, action, null, nonce, isMainnet);
-    const result = await sendRequest(action, signature, nonce, isMainnet);
-
-    if (result.status === 'ok') {
-        return { success: true, data: result };
-    } else {
-        if (result.response && result.response.includes('does not exist')) {
-            printWarn('API Response', 'Account not found, continuing without block size switch');
-            return { success: false, error: result.response || result };
-        } else {
-            printError('API Response Error', result.response || result);
-            throw new Error(`Block size switch failed: ${result.response || result}`);
-        }
-    }
+    const signature = await signL1Action(wallet, action, null, nonce, isMainnet, chain);
+    const result = await sendRequest(action, signature, nonce, chain);
+
+    return result.status === 'ok'
+        ? { success: true, data: result }
+        : (() => {
+              throw new Error(result.response || result);
+          })();
```

Comment:
```suggestion
    if (result.status !== 'ok') {
        throw new Error(result.response || result);
    }

    return result;
```
use the fail-early/return early on negative cases pattern so the happy path is always easy to read

Reply:
and no need to return `success` in an object. failure would cause an exception now